### PR TITLE
Feature/add authorization

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
### Project requirements
- [x] Install CanCanCan in your project.
- [x] Add a role column to the user's table. Remember to use a migration for this.
- [x] A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
   - For that you need to implement the post-deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- [x] A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
    - 
image
image

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.admin?
+      can :manage, :all
+    else
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
+    end
```
Comment (VICTOR-CHUKWU) [inline app/models/ability.rb]: - [ ] you have done a great job authorizing users, however in this snippet, you are giving none admin users the ability to only delete posts and comments that they created, however, you are missing giving them the ability to update and create posts and comments and also the ability to read all, you can make your code to look like this 
```
can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -9,11 +9,12 @@ def index
   # The function assigns variables for a post, comment, like, and user in order
   # to display them in a view.
   def show
-    @post = Post.find(params[:id])
+    @user = current_user
+    @post = Post.find_by(id: params[:id])
```
Comment (VICTOR-CHUKWU) [inline app/controllers/posts_controller.rb:15]: - [ ] add this `load_and_authorize_resource` at the top of your class to authorize posts

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Benawi  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- can cancan ability correctly used
- admin role properly defined


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Benawi** [comment]
### Changes ♻️
- [x] Update user authorizing 
- [x] Add `load_and_authorize_resourc`e at the top of the class to authorize posts

**Review by TadesseDev — APPROVED** [review]
### Hi @Benawi 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**